### PR TITLE
feat: initial commit

### PR DESCRIPTION
**Pull request by krishna2020**
**Fixes** # (*issue*)
Initial commit for custom pixel code that can be embedded in shopify store to forward events to rudderstack

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+          __globalThis_magic__.globalThis = __globalThis_magic__;
+          delete Object.prototype.__globalThis_magic__;
+        }
+        window.rudderAnalyticsAddScript(
+          ""
+            .concat(sdkBaseUrl, "/")
+            .concat(window.rudderAnalyticsBuildType, "/")
+            .concat(sdkName),
+          "data-rsa-write-key",
+          "2fvTvK3sVF1ZZbB73Ov6PmJjpLt",
```
Comment (saikumarrs) [inline src/index.js]: This should be replaced with placeholder write key.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+## (2024-10-10)
+
+### Features
+
+* Initial commit of custom pixel for sending subscribed client-side events to Rudderstack
```
Comment (saikumarrs) [inline CHANGELOG.md]:
```suggestion
* Initial commit of custom pixel for sending subscribed client-side events to RudderStack
```

File: README.md [hunk 3]
```diff
@@ -33,11 +34,8 @@
 ## Getting started
 
 \*\*Describe how to use the software.\*\*
+Replace all the references of the DATAPLANE_URL & WRITE_KEY placeholder with actual values
```
Comment (saikumarrs) [inline README.md]: Let's also point to the file.

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,636 @@
...
+          __globalThis_magic__.globalThis = __globalThis_magic__;
+          delete Object.prototype.__globalThis_magic__;
+        }
+        window.rudderAnalyticsAddScript(
+          ""
+            .concat(sdkBaseUrl, "/")
+            .concat(window.rudderAnalyticsBuildType, "/")
+            .concat(sdkName),
+          "data-rsa-write-key",
+          "<WRITE_KEY>",
```
Comment (saikumarrs) [inline src/index.js]: We can reference the variable directly.
```suggestion
          "WRITE_KEY",
```

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,636 @@
...
+  { shopifyField: "variant.product.id", rudderField: "product_id" },
+  { shopifyField: "variant.product.title", rudderField: "variant" },
+  { shopifyField: "variant.product.type", rudderField: "category" },
+  { shopifyField: "variant.product.url", rudderField: "url" },
+  { shopifyField: "variant.product.vendor", rudderField: "brand" },
+];
+
+(function () {
+  "use strict";
+  window.RudderSnippetVersion = "3.0.14";
```
Comment (saikumarrs) [inline src/index.js]: The snippet is a bit old. Can we use the latest one from the dashboard?

File: README.md (hunk 3 above)
Comment (saikumarrs) [inline README.md]: The placeholders are in fact, `<DATA_PLANE_URL>` and `<WRITE_KEY>`.

File: README.md
Comment (saikumarrs) [inline README.md:1]: Don't we need to add any instructions on how to use it in Shopify custom pixel?

File: README.md [hunk 6]
```diff
@@ -23,6 +23,7 @@
 \*\*Repo description\*\*
 
 ## Overview
+This project provides an easy-to-follow guide and implementation for adding custom pixels to Shopify stores, allowing you to subscribe to client-side events like page views, product clicks, add-to-cart, and purchases for forwarding it to Rudderstack.
```
Comment (saikumarrs) [inline README.md]:
```suggestion
This project provides an easy-to-follow guide and implementation for adding custom pixels to Shopify stores, allowing you to subscribe to client-side events like page views, product clicks, add-to-cart, and purchases for forwarding it to RudderStack.
```

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: README.md
Comment (krishna2020) [reply README.md:1]: Added the shopify doc link

File: README.md [hunk 7]
```diff
@@ -33,11 +34,9 @@
 ## Getting started
 
 \*\*Describe how to use the software.\*\*
+Replace all the references of the <DATAPLANE_URL> & <WRITE_KEY> placeholder with actual values at (src/index.js)
+Follow https://help.shopify.com/en/manual/promoting-marketing/pixels/custom-pixels/manage?shpxid=6c5e6078-ECEC-4AC3-EE41-151B603831FB for adding the custom pixel in the store
 
 ## Contribute
```
Comment (saikumarrs) [inline README.md]: This section should also be removed given that we don't have this file.

File: src/index.js (hunk 5 above)
Comment (krishna2020) [reply src/index.js]: @yashasvibajpai can you get latest snippet from source setup page in webapp & test if everything is intact ?

File: README.md (hunk 7 above)
Comment (yashasvibajpai) [reply README.md]: I think this is correct, it is pointing to the custom pixel code in the src folder here

File: src/index.js (hunk 5 above)
Comment (yashasvibajpai) [reply src/index.js]: updated

File: README.md (hunk 7 above)
Comment (krishna2020) [reply README.md]: I think @saikumarrs meant Contribute section

Comment (saikumarrs) [reply README.md]: My comment is intended for the "Contribute" section.

Comment (saikumarrs) [reply README.md]: GitHub does not properly highlight the selected line 👎

File: src/index.js [hunk 8]
```diff
@@ -281,35 +254,36 @@ const checkoutStartedCompletedEventMapping = [
...
-        window.rudderAnalyticsAddScript(
-          "https://polyfill-fastly.io/v3/polyfill.min.js?version=3.111.0&features=Symbol%2CPromise&callback=rudderAnalyticsMount",
-        );
+        window.rudderAnalyticsAddScript("https://polyfill-fastly.io/v3/polyfill.min.js?version=3.111.0&features=Symbol%2CPromise&callback=rudderAnalyticsMount");
       } else {
         window.rudderAnalyticsMount();
       }
       var loadOptions = {};
-      rudderanalytics.load(WRITE_KEY, DATAPLANE_URL, loadOptions);
+        rudderanalytics.load(WRITE_KEY, DATAPLANE_URL, loadOptions);
```
Comment (saikumarrs) [inline src/index.js]: I'll fix this in the snippet for all later. Can we please fix it here?
```suggestion
      rudderanalytics.load(WRITE_KEY, DATAPLANE_URL, loadOptions);
```

Comment (yashasvibajpai) [reply src/index.js]: fixed

File: README.md (hunk 7 above)
Comment (yashasvibajpai) [reply README.md]: got it, updated

**Review by saikumarrs — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**